did/interface.go

```diff
@@ -0,0 +1,23 @@
+package did
+
+import "strings"
+
+// TODO: All DID's MUST do X?
+// Define the interface more clearly
+type DID interface {
+	IsValid() bool
+	Resolve() (*DIDDocument, error)
```

Comment:
hmm, not sure about this one. 
some DIDs (like ION, and others) require network resolvers, which require hosted software. I'd like to avoid making recommendations of which resolvers to use here.

some DIDs (like peer, web, key) can be resolved more simply.

Reply:
Sure. I'm not sure about this one either. I do think we need a DID interface though! I'm just not sure exactly what goes into it. IsValid() seems reasonable. What else is required in a DID?

Reply:
hmm...maybe some common methods to get keys? like `getAuthenticationMethods`, `getVerificationMethods` and so on?

would validation just be static, or should we eventually expand to determining whether the keys in the doc are themselves valid?

there also needs to be a more flexible way to determine validity across DID methods, as they each have a set of properties/processes that uniquely apply to each method

Reply:
I think that's why each DID needs to implement an IsValid method no? That method would determine if the DID is valid or not. 

It seems like that the biggest issue here is the network resolvers?

Reply:
For resolution, here's what I ended up doing which works: 

```
type Resolver interface {
	Resolve(d DID) (*DIDDocument, error)
}
```

Some did's will have resolvers built in, and so will not. So network based resolvers we can ignore fine, it just won't be implementing the Resolver interface